overlays/bootstrap.nix

```diff
@@ -453,11 +447,36 @@ in {
           done
         '';
       };
+
+    # Memoize the cabal-install and nix-tools derivations by adding:
+    #   haskell-nix.cabal-install.ghcXXX
+    #   haskell-nix.cabal-install-unchecked.ghcXXX
+    #   haskell-nix.nix-tools.ghcXXX
+    #   haskell-nix.nix-tools-unchecked.ghcXXX
+    # Using these avoids unecessary calls to mkDerivation.
+    # For cabal projects we match the versions used to the compiler
+    # selected for the project to avoid the chance of a dependency
+    # another GHC version (particularly useful on macOS where
+    # executables are dynamically linked to GHC itself).
```

Comment:
... which means that if you use a tool built with a different GHC you will get that GHC itself in your closure.